swiftsimio/reader.py

```diff
@@ -16,16 +16,28 @@
 from swiftsimio.objects import cosmo_array, cosmo_factor, a
 from swiftsimio.conversions import swift_cosmology_to_astropy
 
+import cloudpickle
 import re
 import h5py
+import json
 import unyt
 import numpy as np
+import numpy.typing as npt
+import requests
 import warnings
 
 from datetime import datetime
 from pathlib import Path
 
 from typing import Union, Callable, List, Optional
+from pathlib import Path
+
+
+class NumpyEncoder(json.JSONEncoder):
+    def default(self, obj):
+        if isinstance(obj, np.ndarray):
+            return obj.tolist()
+        return json.JSONEncoder.default(self, obj)
```

Comment:
In general it would be great to have all the remote stuff in a separate file (e.g. remote.py)